app/services/decisions/dwp_decision_tree.rb

```diff
@@ -46,19 +49,37 @@ def after_confirm_details
     end
 
     def after_benefit_type
+      return edit(:partner_benefit_type) if partner_benefit_type_required?
+
+      benefit_check_routing(applicant)
+    end
+
+    def after_partner_benefit_type
+      benefit_check_routing(partner)
+    end
+
+    def benefit_check_routing(person)
       if form_object.benefit_type.none?
         if FeatureFlags.means_journey.enabled?
           edit('/steps/case/urn')
         else
           show(:benefit_exit)
         end
-      elsif !applicant_has_nino
+      elsif !has_nino(person)
         edit(:cannot_check_benefit_status)
       else
-        determine_dwp_result_page
+        determine_dwp_result_page(person)
       end
     end
 
+    def partner_benefit_type_required?
+      # TODO: refactor to use has_partner in partner_details table instead of client_has_partner
+
+      form_object.benefit_type.none? &&
```

Comment:
I think it would be good to do the task list side of this next. I suspect quite a bit of this logic will need to be shared with the answers validator.